repoparse/repoparse.py

```diff
@@ -0,0 +1,422 @@
+        next_stage_repos=2, pool_processes=5
+) -> Tuple[Union[None, List[str]], Union[ParsedRepo, List[ParsedRepo]]]:
+    """Main repo parse process entrance point"""
+    absolute_out_path = path.join(base_path, out_dir_name)
+    absolute_build_path = path.join(base_path, build_dir_name)
+    repos_for_next_parse: Union[None, List[str]] = None
+
+    if not path.exists(absolute_out_path):
+        makedirs(absolute_out_path)
+    if not path.exists(absolute_build_path):
+        makedirs(absolute_build_path)
+
+    is_remote = repo_path.startswith("http") or repo_path.startswith("git@")
+
+    if not path.exists(repo_path) and not is_remote:
+        raise ValueError("Unknown repo...")
+
+    if is_remote:
+        repos_for_next_parse = calculate_stars(
+            repo_path, git_key_path, parse_depth, max_stargazers,
+            max_stargazers_starred, next_stage_repos
+        )
+        repo_path = clone_and_detect_path(
+            repo_path, do_reclone, absolute_build_path
+        )
+
+    repo_path = path.join(base_path, repo_path)
+
+    print("Doing the enry part...")
+    enry_on_latest_revision = run_enry_by_file(base_path, repo_path)
+
+    print("Parsing the repo...")
+    result = parse_repo_commits(
+        repo_path, enry_on_latest_revision, is_quiet, absolute_out_path, 
+        absolute_build_path, chunked=is_chunked, chunk_size=chunk_size, 
+        pool_processes=pool_processes
+    )
+    if repos_for_next_parse:
+        return repos_for_next_parse, result
+    else:
+        return None, result
+
+
+class RepoProcessWorker(multiprocessing.Process):
+    """This is an abstraction for a process that works to parse a repo"""
+
+    def __init__(
+        self, *, base_path: str, repo_path: str, do_reclone: bool, 
+        is_quiet: bool, is_chunked: bool, chunk_size: int, 
+        visited_dict: MutableMapping[str, int], visited_lock: threading.Lock,
+        out_dir_name="out", build_dir_name="build", 
+        git_key_path="build/github_key", parse_depth=3, max_stargazers=1000, 
+        max_stargazers_starred=10, next_stage_repos=4, pool_processes=5
+    ):
+        super().__init__()
+        self.base_path = base_path
+        self.repo_path = repo_path
+        self.do_reclone = do_reclone
+        self.is_quiet = is_quiet
+        self.is_chunked = is_chunked
+        self.chunk_size = chunk_size
+        self.out_dir_name = out_dir_name
+        self.build_dir_name = build_dir_name
+        self.git_key_path = git_key_path
+        self.parse_depth = parse_depth  # Next parse step...
+        self.max_stargazers = max_stargazers
+        self.max_stargazers_starred = max_stargazers_starred
+        self.next_stage_repos = next_stage_repos
+        self.visited_dict = visited_dict
+        self.visited_lock = visited_lock
+        self.pool_processes = pool_processes
+
+    def run(self):
+        print(f"PARSE DEPTH: {self.parse_depth}")
+
+        try:
+            next_repos, r = parse_repo(
```

Comment:
`r` is really bad name - please give a meaningful name

Reply:
it's even harder to understand because you don't have docstring that explains what returned `result` should contain